src/main/java/com/twozo/smapp/controller/MessageController.java

```
@@ -0,0 +1,141 @@
+package com.twozo.smapp.controller;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import com.twozo.smapp.model.ApiResponse;
+import com.twozo.smapp.model.InboxInfo;
+import com.twozo.smapp.model.Message;
+import com.twozo.smapp.model.User;
+import com.twozo.smapp.service.MessageService;
+import com.twozo.smapp.validation.MessageValidator;
+import com.twozo.smapp.validation.ValidationType;
+
+@RestController
+@RequestMapping("/api/message")
+public class MessageController {
+
+    private final MessageService messageService;
+    private final MessageValidator messageValidator;
+
+    public MessageController(final MessageService messageService, final MessageValidator messageValidator) {
+        this.messageService = messageService;
+        this.messageValidator = messageValidator;
+    }
+
+    @PostMapping("/send")
+    public ResponseEntity<?> send(@RequestBody final Message message) {
+        message.setSentTimestamp((Instant.now()));
+        final Collection<String> errors = messageValidator.validate(message, ValidationType.ADD);
+
+        if (!errors.isEmpty()) {
+            return ResponseEntity.badRequest().body(new ApiResponse(errors.toString()));
+        }
+
+        final boolean result = messageService.send(message);
+
+        if (result) {

```

Instead of assuming that a false result indicates the sender or receiver details are not found, throw appropriate exceptions, handle them properly, and include them in the response.